src/DETHRACE/pc-win95/win95net.c

```diff
@@ -0,0 +1,676 @@
+int gMatts_PC;
+tU16 gSocket_number_pd_format;
+//_IPX_ELEMENT gListen_elements[16];
+char gLocal_ipx_addr_string[32];
+//_IPX_ELEMENT gSend_elements[16];
+struct sockaddr_in* gLocal_addr_ipx;
+char gReceive_buffer[512];
+tPD_net_player_info gRemote_net_player_info;
+struct sockaddr_in* gBroadcast_addr_ipx;
+tPD_net_player_info gLocal_net_player_info;
+char gSend_buffer[512];
+tIPX_netnum gNetworks[16];
+struct sockaddr_in* gRemote_addr_ipx;
+tU8* gSend_packet;
+// W32 gListen_segment;
+tU8* gSend_packet_ptr;
+// W32 gSend_segment;
+tU8* gListen_packet;
+tU8* gListen_packet_ptr;
+size_t gMsg_header_strlen;
+int gNumber_of_networks;
+int gNumber_of_hosts;
+//_IPX_HEADER gLast_received_IPX_header;
+tU16 gSocket_number_network_order;
+// unsigned short gECB_offset;
+tU16 gListen_selector;
+tU16 gSend_selector;
+
+struct sockaddr_in gLocal_addr;
+struct sockaddr_in gRemote_addr;
+struct sockaddr_in gBroadcast_addr;
+struct sockaddr_in gLast_received_addr;
+int gSocket;
+
+#define MESSAGE_HEADER_STR "CW95MSG"
+#define JOINABLE_GAMES_CAPACITY 16
+
+// IDA: void __cdecl ClearupPDNetworkStuff()
+void ClearupPDNetworkStuff(void) {
+    LOG_TRACE("()");
+    NOT_IMPLEMENTED();
+}
+
+// IDA: void __usercall MATTMessageCheck(char *pFunction_name@<EAX>, tNet_message *pMessage@<EDX>, int pAlleged_size@<EBX>)
+void MATTMessageCheck(char* pFunction_name, tNet_message* pMessage, int pAlleged_size) {
+    LOG_TRACE("(\"%s\", %p, %d)", pFunction_name, pMessage, pAlleged_size);
+    NOT_IMPLEMENTED();
+}
+
+// IDA: int __usercall GetProfileText@<EAX>(char *pDest@<EAX>, int pDest_len@<EDX>, char *pFname@<EBX>, char *pKeyname@<ECX>)
+int GetProfileText(char* pDest, int pDest_len, char* pFname, char* pKeyname) {
+    FILE* fp;
+    char in_buf[256];
+    int i;
+    int j;
+    int len;
+    LOG_TRACE("(\"%s\", %d, \"%s\", \"%s\")", pDest, pDest_len, pFname, pKeyname);
+    NOT_IMPLEMENTED();
+}
+
+// IDA: int __cdecl GetSocketNumberFromProfileFile()
+int GetSocketNumberFromProfileFile(void) {
+    char str[256];
+    int sscanf_res;
+    tU32 socknum;
+    LOG_TRACE("()");
+    NOT_IMPLEMENTED();
+}
+
+// tU32 EthernetAddressToU32(SOCKADDR_IPX_* pAddr_ipx) {
+//     LOG_TRACE("(%p)", pAddr_ipx);
+//     NOT_IMPLEMENTED();
+// }
+
+void NetNowIPXLocalTarget2String(char* pString, struct sockaddr_in* pSock_addr_ipx) {
+    LOG_TRACE("(\"%s\", %p)", pString, pSock_addr_ipx);
+
+    char portbuf[10];
+
+    inet_ntop(AF_INET, &pSock_addr_ipx->sin_addr, pString, 32);
+    sprintf(portbuf, ":%d", ntohs(pSock_addr_ipx->sin_port));
+    strcat(pString, portbuf);
+}
+
+// IDA: int __usercall GetMessageTypeFromMessage@<EAX>(char *pMessage_str@<EAX>)
+int GetMessageTypeFromMessage(char* pMessage_str) {
+    char* real_msg;
+    int msg_type_int;
+    LOG_TRACE("(\"%s\")", pMessage_str);
+
+    real_msg = &pMessage_str[4];
+    msg_type_int = 0;
+
+    // FIXME: "CW95MSG" value is used in and depends on platform
+    if (strncmp(real_msg, "CW95MSG", gMsg_header_strlen) == 0) {
```

Comment:
The comment should probably move to another location.
```suggestion
    // FIXME: "CW95MSG" value is used in and depends on platform
    if (strncmp(real_msg, MESSAGE_HEADER_STR, gMsg_header_strlen) == 0) {
```